images/kindnetd/cmd/kindnetd/cni.go

```diff
@@ -33,22 +33,33 @@ import (
 
 // CNIConfigInputs is supplied to the CNI config template
 type CNIConfigInputs struct {
-	PodCIDR      string
-	DefaultRoute string
-	Mtu          int
+	PodCIDRs      []string
+	DefaultRoutes []string
+	Mtu           int
 }
 
 // ComputeCNIConfigInputs computes the template inputs for CNIConfigWriter
 func ComputeCNIConfigInputs(node corev1.Node) CNIConfigInputs {
-	podCIDR := node.Spec.PodCIDR
-	defaultRoute := "0.0.0.0/0"
 
-	if net.IsIPv6CIDRString(podCIDR) {
-		defaultRoute = "::/0"
+	defaultRoutes := []string{"0.0.0.0/0", "::/0"}
+	// check if is a DualStack cluster
+	if len(node.Spec.PodCIDRs) > 1 {
```

Comment:
this will be zero on older clusters? or some client side magic.

Reply:
since PodCIDRs is a []string , my understanding is that client-go will set the nil value if it does not exist in the server.
You know that @liggitt will illustrate us :smile: